File: 프로그래머스/1/42840. 모의고사/모의고사.js
function compute(answers, player) {
    let hit = 0;
    for(let i = 0; i < answers.length; i++) {
        if(answers[i] === player[i % player.length]) {
            hit++;
        }
    }
    return hit;
}


function solution(answers) {
    let max = 0;
    let answer = [];
    const hits = [];
    const player1 = [1, 2, 3, 4, 5];
    const player2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const player3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    hits.push(compute(answers, player1));
    hits.push(compute(answers, player2));
    hits.push(compute(answers, player3));
    
    max = Math.max(...hits)
    for(let i = 0; i < 3; i++) {
        if(hits[i] === max) {
            answer.push(i + 1)
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def compute_correct_num(my_answer, answer):
    cycle = len(my_answer)
    correct = 0
    for idx, num in enumerate(answer):
        my_pick = my_answer[idx % cycle]
        if my_pick == num:
            correct += 1

    return correct
        

def solution(answers):
    answer = []
    supoja = [
        [1,2,3,4,5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    corrects = []
    for i in range(3):
        corrects.append(compute_correct_num(supoja[i], answers))
    
    max_correct = max(corrects)
    for i in range(3):
        if corrects[i] == max_correct:
            answer.append(i + 1)
            
    return answer
        